app/models/manageiq/providers/google/inventory/parser.rb

```diff
@@ -340,6 +340,8 @@ def get_os_product_name(storage)
   def parse_instance_parent_image(instance)
     parent_image_uid = nil
 
+    return parent_image_uid if instance.disks.empty?
+
     instance.disks.each do |disk|
       parent_image_uid = @cloud_volume_url_to_source_image_id[disk[:source]]
       next if parent_image_uid.nil?
```

Comment:
This is an awkward way to write this (not your change, the original method).
The next if .nil? then break always instead of `break if .present?`
Setting a local var in the loop and then not using it other than to return it, could just return it directly in this loop.

I think this whole method can be simply one line:
`instance.disks&.detect { |disk| @cloud_volume_url_to_source_image_id[disk[:source]] }`